src/library/commissioner_safe.cpp

```diff
@@ -107,13 +106,23 @@ Error CommissionerSafe::Start()
 // Stop the commissioner running in background.
 void CommissionerSafe::Stop()
 {
-    mImpl.Stop();
+    std::promise<void> pro;
 
-    if (mEventThread && mEventThread->joinable())
-    {
-        mEventThread->join();
-        mEventThread = nullptr;
-    }
+    VerifyOrExit(mEventThread.joinable());
```

Comment:
Multiple threads can pass `VerifyOrExit` at the same time?

Reply:
it is not safe to be called from multiple user threads. mark this as resolved becuase we lowered the thread safety of this `CommissionerSafe` to be only safe between one single user thread and the event loop thread.